### DISCUSSION TITLE
A few questions regarding advanced settings

### DISCUSSION THREAD

**Discussion by blogdemoi** [mcguirepr89/BirdNET-Pi#507] [Q&A]
> When the disk becomes full, you can choose to 'purge' old files to make room for new ones or 'keep' your data and stop all services instead.
> Note: you can exclude specific files from 'purge' on the Recordings page.

Does it concern the audio files and the database or only the audio files? As I wish to keep the database for long term analysis of birds in my garden.

> Set Channels to the number of channels supported by your sound card. 32 max.

How can I check the number of channels supported by my sound card?

What is "Silence Update Indicator"?

Below "BirdNET-Lite Settings" what are "Overlap", "Minimum Confidence", "Sigmoid Sensitivity"?

Thank you.

**Comment by ehpersonal38** [comment]
> Does it concern the audio files and the database or only the audio files? As I wish to keep the database for long term analysis of birds in my garden.

Only extracted audio clips are affected by the purging. Your database will always remain unharmed and should last for many years, though regular backups are recommended.

> What is "Silence Update Indicator"?

There's a little feature that whenever your BirdNET-Pi is 50 commits or more behind the Github repo, a bit of red text will show next to the "Tools" link that will say "50+," just to let you know it's probably time to update to get the latest bug fixes and whatnot. That option in advanced is just for disabling that red text.

Not confident on the other questions, maybe someone else can help.

**Comment by CaiusX** [comment]
Hi @blogdemoi 

Regarding your last question:

Overlap - whether the samples sent for analysis overlap time wise - so if your sample length is 3s and overlap is 1.5s, 3 second samples will be sent starting at time time 1.5 and time 3, basically doubling the amount of analysis - illustrated below: 
![image](https://user-images.githubusercontent.com/25373276/185737367-1f9330cc-b7ed-49f1-8d9c-c95b1ec3d5be.png)
I generally keep it at zero for space conservation, at the risk of missing identifications spanning sample end - begin next sample

Minimum Confidence is the confidence above which identification is posted to the database and sound file saved.  Play with this (together with your "Excluded Species" list) to find a sweet spot of true positive identifications without any (or minimal) false positives.  
I generally keep this around 0.5.

I'm not clear on sigmoid sensitivity, but also affects the probability of a true/false positive.  I keep this constant at 1.25 on all my systems.  

Best
AP

**Reply by blogdemoi** [reply]
Thank you.

**Comment by DCS17** [comment]
Regarding Sigmoid Sensitivity: 
in the original BirdNet of which BirdNetPi is a fork it says this is an argument for analyze.py
--sensitivity: Sigmoid sensitivity; Higher values result in lower sensitivity. Values in [0.25, 2.0]. Defaults to 1.0.
I'm in the process of seeing what difference it makes to identifications by changing every couple of days.
Currently using 1.0

**Reply by blogdemoi** [reply]
Thank you.

**Reply by Fish31520** [reply]
Hi @DCS17,

Have you tried messing with sigmoid sensitivity ? Which values did you use and for which result ?

Thanks !

**Reply by DCS17** [reply]
Hi @Fish31520 ,
Yes I have messed about with it. In all honesty I didn't see much difference. So I'm back to using 1.0 !

**Reply by Fish31520** [reply]
Thanks @DCS17 for your feedback !

Mine is currently tuned at 1.25, so I think I'll leave it like that.

Have a nice day !

**Reply by arne1921KF** [reply]
Having read the [information kindly provided on the technical aspects](https://github.com/mcguirepr89/BirdNET-Pi/wiki/BirdNET-Pi:-some-theory-on-classification-&-some-practical-hints) by @DD4WH and @ehpersonal38 I still wasn't sure about the effects of the sigmoid sensitivity setting. 

I turned to the original source. In [Kahl et al. (2021) Ecol Inf 61:101236](https://www.sciencedirect.com/science/article/pii/S1574954121000273), they write:

_The model sensitivity was empirically set to 0.85 (instead of 1.0), which flattened the sigmoid activation curve in the last layer and required less activation to achieve a score above the detection threshold._ 

Further cursory reading on the sigmoid activation function used in DNNs (e.g. on [some](https://www.v7labs.com/blog/neural-networks-activation-functions) [web](https://towardsdatascience.com/activation-functions-neural-networks-1cbd9f8d91d6) [search hits](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://en.m.wikipedia.org/wiki/Sigmoid_function) makes me think that a if I choose a lower value (minimum is 0.5) in the GUI, the sensitivity of the model is _higher_ in the sense that I would receive more false positives. The algorithm will, like I understand what @kahst and colleagues write in their paper, achieve a higher probability to achieve a score beyond the threshold I have set. A higher sigmoid sensitivity value set in the GUI (maximum is 1.5) would mean that the probability that the threshold is met would be lower, hence the algorithm would be less _sensitive_ to bird calls. 

I'm currently playing around with the minimum and maximum value. So far, it seems my interpretation could be correct. I get far more detections with a sigmoid sensitivity set to low values. FTR, I am only playing with the sigmoid sensitivity value, i.e. without changing the minimum confidence score (in my understanding: the specifity), the recording overlap and the Species Range Model threshold (using Species Range Model V2.4 - V2, since I now switched sources to @Nachtzuster's [fork](https://github.com/Nachtzuster/BirdNET-Pi)).

This seems to contradict the technical information i linked above. Would anyone mind to comment on that? Since this is linked via the GUI, could we perhaps change the technical information?

**Reply by jdchristensen** [reply]
The sigmoid sensitivity setting changes the shape of the function that is used to map the model output to the range [0, 1] (see attached image).  A lower sensitivity means that more of the results will be close to 0.5, and fewer will be near 0 and 1, while a higher sigmoid sensitivity means that more will be near 0 and 1, while fewer will be near 0.5.  If your minimum confidence score is less than 0.5, then a lower sigmoid sensitivity will result in more detections, as @arne1921KF noticed, since outputs that were closer to 0 end up closer to 0.5.  But if your minimum confidence score is greater than 0.5, then a lower sigmoid sensitivity will result in *fewer* detections, since scores that were near 1 get moved to being closer to 0.5.

In terms of which measurements cross the confidence threshold, a change in the sigmoid sensitivity is always equivalent to a change in the minimum confidence, so the minimum confidence is the main thing that users should adjust.

![sigmoid](https://github.com/user-attachments/assets/8816135f-aa0a-447a-af6d-62cfa30bf3c1)

**Reply by jdchristensen** [reply]
To complicate matters, BirdNET-Pi has code that flips the interval [0.5, 1.5] around, exchanging 0.5 and 1.5 and effectively reversing the meaning of the sensitivity parameter!  It seems pretty odd to do this and not document it in the web settings GUI...  Here are links showing the relevant lines of code in this repo and the Nachtzuster fork:
https://github.com/mcguirepr89/BirdNET-Pi/blob/f8e74cad351093f098b4e11d2fe4ef8ba6bc3482/scripts/server.py#L457
https://github.com/Nachtzuster/BirdNET-Pi/blob/e3eb0943b409770a491acf0546256b541915cc83/scripts/server.py#L247

As a result, what I wrote in the previous comment needs to be reversed, with a *higher* BirdNET-Pi sensitivity meaning that the results tend to be closer to 0.5...

**Reply by DMontgomery40** [reply]
That's really confusing so I reversed mine with this command:
```
cd ~/BirdNET-Pi/scripts && \
sudo cp server.py server.py.bak && \
sudo sed -i 's/sensitivity = max(0.5, min(1.0 - (args.sensitivity - 1.0), 1.5))/sensitivity = max(0.5, min(args.sensitivity, 1.5))/' server.py && \
echo "## patched sigmoid inversion ##" && grep -n "sensitivity =" server.py && \
sudo systemctl restart birdnet-pi```
